pkg/proxy/proxy.go

```diff
@@ -241,22 +245,24 @@ func (p *Proxy) RoundTrip(req *http.Request) (*http.Response, error) {
 }
 
 func (p *Proxy) tokenReview(req *http.Request) (*http.Response, error) {
-	klog.V(4).Infof("attempting to validate a token in request using TokenReview endpoint(%s)",
-		req.RemoteAddr)
+	// Get the source client address of request if x-forwarded.
+	clientAddress := xff.GetRemoteAddr(req)
```

Comment:
It seems like every method that ever uses the remote address will need to explicitly parse it from the request - is this expected/desirable, or would it be better to pass it as an arg to this method? Are there many places this needs to happen?

Reply:
Going to a http.Handler paradigm this will become a bit of a requirement.
It is also not particularly expensive:
```

// GetRemoteAddr parses the given request, resolves the X-Forwarded-For header
// and returns the resolved remote address.
func GetRemoteAddr(r *http.Request) string {
	return GetRemoteAddrIfAllowed(r, func(sip string) bool { return true })
}

// GetRemoteAddrIfAllowed parses the given request, resolves the X-Forwarded-For header
// and returns the resolved remote address if allowed.
func GetRemoteAddrIfAllowed(r *http.Request, allowed func(sip string) bool) string {
	if xffh := r.Header.Get("X-Forwarded-For"); xffh != "" {
		if sip, sport, err := net.SplitHostPort(r.RemoteAddr); err == nil && sip != "" {
			if allowed(sip) {
				if xip := Parse(xffh); xip != "" {
					return net.JoinHostPort(xip, sport)
				}
			}
		}
	}
	return r.RemoteAddr
}
```

Reply:
Kind of annoying, I wrote a whole reply to this that has gone missing :sweat_smile: 

Basically, the function call is not that expensive. 
```// GetRemoteAddr parses the given request, resolves the X-Forwarded-For header
// and returns the resolved remote address.
func GetRemoteAddr(r *http.Request) string {
	return GetRemoteAddrIfAllowed(r, func(sip string) bool { return true })
}

// GetRemoteAddrIfAllowed parses the given request, resolves the X-Forwarded-For header
// and returns the resolved remote address if allowed.
func GetRemoteAddrIfAllowed(r *http.Request, allowed func(sip string) bool) string {
	if xffh := r.Header.Get("X-Forwarded-For"); xffh != "" {
		if sip, sport, err := net.SplitHostPort(r.RemoteAddr); err == nil && sip != "" {
			if allowed(sip) {
				if xip := Parse(xffh); xip != "" {
					return net.JoinHostPort(xip, sport)
				}
			}
		}
	}
	return r.RemoteAddr
}
```

It also comes a bit of requirement a requirement to do it this way when going to the Handler implementation. In theory we could hold it in context though. Wdyt? @munnerz

Reply:
My concern was more around different implementations being used/if use of this were to ever change, you'd have a lot of call-sites to consider. But happy to go for it as it is if you think it's okay 😄